flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/HsSubpartitionMemoryDataManager.java

```diff
@@ -145,8 +146,19 @@ public Optional<HsMemoryDataManager.BufferAndNextDataType> consumeBuffer(int toC
                                 tuple.f0.getBufferIndexAndChannel()));
         return bufferAndNextDataType.map(
                 tuple ->
-                        new HsMemoryDataManager.BufferAndNextDataType(
-                                tuple.f0.getBuffer(), tuple.f1));
+                        new BufferAndBacklog(
+                                tuple.f0.getBuffer(), getBacklog(), tuple.f1, toConsumeIndex));
+    }
+
+    // Get backlog of memory data, always return 0.
```

Comment:
Move to inside the method. And explain why.

Reply:
I changed the logical of calculate memory data's backlog, it will return a meaningful value instead 0 now.